### ISSUE TITLE
Attunement script errors

### ISSUE THREAD

**Issue by BinuDR** [rpherbig/dr-scripts#4485]
The recent change to this script has introduced some problems as noted in my review of the closed PR.

https://github.com/rpherbig/dr-scripts/pull/4482

**Comment by BinuDR** [comment]
@KatoakDR

**Comment by KatoakDR** [comment]
Hi @BinuDR,

Good catch on the timer bug! I didn’t notice it when I was making my changes. That was the original logic so didn’t think to modify it.

It likely wasn’t noticeable before because the script only power walked the rooms once. For lunar mages the script only perceived exactly once so never would have reached the timer limit.

Fixing the timer check should be an easy update. 

> The script needs to handle lunar mages differently. It could be improved with the various options for perceiving that are available (planets, the moons, spellbooks, etc). Implementing more perceive methods for them can be a project for later though.

I agree the script could be doing better job for lunar mages. In my recent update I wasn’t addressing that specifically but rather ensuring the script did what it was doing until the learning rate was met.

> The previous behavior of perceiving once and then exiting was decent.

Could you share why you think that approach was decent? For me as a lunar mage the script only perceived once but never ensured my attunement got locked. As a training script I felt it was not serving its purpose if it only perceived once. That’s why I introduced the loops to ensure the script trained to the desired learning rate. 

Thanks

**Comment by BinuDR** [comment]
There's a 60 second timer on learning from power perceiving so currently it will keep going and you only learn every 1 minute.  Given that, it's a big waste of time to run the script as it currently is.

Perceiving once and stopping worked before without wasting time.  You could do other things during the 1 minute downtime and call the script again afterwards.

**Comment by KatoakDR** [comment]
Thanks for the clarification, I'm going to look at making it more efficient. Possibly support an argument to behave the previous way of "one and done" or a "keep going until mind locked".

**Comment by KatoakDR** [comment]
What's the importance of `Time.now - @start_timer >= 90` ? Why 90 seconds?
